src/main/java/com/hariharanweb/remoteappiummanager/controller/AppiumController.java

```diff
@@ -48,12 +53,13 @@
         return response.body();
     };
 
-    private AppiumDriverLocalService startAppiumServer(String path) {
+    private AppiumDriverLocalService startAppiumServer(String path, int port) {
 
+        AppiumServiceBuilder appiumServiceBuilder = new AppiumServiceBuilder().withAppiumJS(new File(path))
+                .withArgument(GeneralServerFlag.LOG_LEVEL, "info")
+                .withIPAddress("127.0.0.1");
         AppiumServiceBuilder builder =
-                new AppiumServiceBuilder().withAppiumJS(new File(path))
-                        .withArgument(GeneralServerFlag.LOG_LEVEL, "info")
-                        .withIPAddress("127.0.0.1")
+                port > 0 ? appiumServiceBuilder.usingPort(port) : appiumServiceBuilder
```

Comment:
@bsneha90 we don't have to handle this, as appiumServer will start on any port that is free